File: Src/ButterflyNet.Parser/Strategies/LineBreakStrategy.cs
using ButterflyNet.Parser.Scopes;

namespace ButterflyNet.Parser.Strategies {
	[TokenTransformer("%%%")]
	public sealed class LineBreakStrategy : InlineStrategy {
		protected override void DoExecute(ParseContext context) {
			OpenScope(new LineBreakScope(), context);
			CloseCurrentScope(context);
		}
	}
}

File: src/js/src/strategies/LineBreakStrategy.js
function LineBreakStrategy() {
	LineBreakStrategy.$parent.call(this);

	this.setAsTokenTransformer("%%%");

	this.doExecute = function(context) {
		this.openScope(new LineBreakScope(), context);
		this.closeCurrentScope(context);
	};
}

extend(InlineStrategy, LineBreakStrategy);